packages/back-end/src/models/ImpactEstimateModel.ts

```diff
@@ -66,11 +66,7 @@ export async function getImpactEstimate(
 
   let segmentObj: SegmentInterface | null = null;
   if (segment) {
-    segmentObj = await SegmentModel.findOne({
-      id: segment,
-      organization,
-      datasource: datasource.id,
```

Comment:
This query was also filtering by datasource - not sure if there is a reason for that, or if segment and organization alone are sufficient.